asciidoc/courses/genai-mcp-build-custom-tools-python/modules/1-getting-started/lessons/4-mcp-inspector/lesson.adoc

```diff
@@ -0,0 +1,178 @@
+@modelcontextprotocol/inspector@0.17.1
+Ok to proceed? (y) 
+----
+
+Press `y` to proceed.
+
+[TIP]
+.Using `npx`
+====
+The `npx` command is a package runner for Node.js.  It is used to run Node.js packages without installing them globally.
+
+You can install npx along with Node.js by link:https://nodejs.org/en/download[downloading Node.js from the official website^].
+====
+
+Once installed, the MCP Inspector UI will be served on port 6274.
+
+[source]
+----
+Starting MCP inspector...
+⚙️ Proxy server listening on localhost:6277
+🔑 Session token: d3a6bc6d116f75b1a5f7861361fb5fa7190c29b02c70f614e8ccf14080d1eadd
+   Use this token to authenticate requests or set DANGEROUSLY_OMIT_AUTH=true to disable auth
+
+🚀 MCP Inspector is up and running at:
+   http://localhost:6274/?MCP_PROXY_AUTH_TOKEN=d3a6bc6d116f75b1a5f7861361fb5fa7190c29b02c70f614e8ccf14080d1eadd
+
+🌐 Opening browser...
+----
+
+Note the Session token, you may need to copy and paste this into the Configuration tab of the MCP Inspector to connect to your server and run tools.
+
+==== Pre-filling the Server details 
+
+You can pre-fill the details of your server by appending the configuration details to the command.
+
+[source,shell]
+.Running the MCP Inspector with pre-filled server details
+----
+npx @modelcontextprotocol/inspector --url http://localhost:8000/mcp --transport http
+----
+
+// This will run the `main.py` file in the `server` directory using the `uv` command as we instructed VS Code to do in the previous lesson.
+
+// [TIP]
+// .`$PWD`
+// ====
+// The `$PWD` environment variable is used to get the current working directory, which is required by the `uv` command.
+
+// The command assumes that you are running the command from the root of the project.
+// If you are running the command from the `server/` directory you can omit the directory name from the command.
+// ====
+
+
+// Once you have opened the UI and verified the Transport Type is `stdio`, you can click the **Connect** button to connect to your server.
+
+
+
+== Using the MCP Inspector
+
+// To connect to your server, you will need to either specify the full path of your file, or switch the transport method to HTTP.
+
+// [source,shell]
+// ----
+// npx @modelcontextprotocol/inspector
+// Starting MCP inspector...
+// ⚙️ Proxy server listening on 127.0.0.1:6277
+// 🔑 Session token: 32be7bf018a86d10c0428db91e0ff4ad32236a664e176642451b1ebbcaf69869
+// Use this token to authenticate requests or set DANGEROUSLY_OMIT_AUTH=true to disable auth
+
+// 🔗 Open inspector with token pre-filled:
+//    http://localhost:6274/?MCP_PROXY_AUTH_TOKEN=32be7bf018a86d10c0428db91e0ff4ad32236a664e176642451b1ebbcaf69869
+
+// 🔍 MCP Inspector is up and running at http://127.0.0.1:6274 🚀
+// ----
+
+Clicking the link in the console will open the MCP Inspector in your browser with the server details pre-filled on the left hand side.  Verify the connection details and click the **Connect** button to connect to your server.
```

Comment:
Suggest adding instructions here on how to connect to the server. Its an important part of the process. Missing it out / pre-filling wont build resiliency and they may need to change them.